core-backend/common/src/lib.rs

```diff
@@ -139,39 +133,22 @@ pub trait Environment<E: Ext + IntoExtInfo + 'static>: Sized {
     /// An error issues in environment.
     type Error: fmt::Display;
 
-    /// Creates new external environment to execute wasm binary:
     /// 1) Instantiates wasm binary.
-    /// 2) Creates wasm memory with filled data (exception if lazy pages enabled).
-    /// 3) Instantiate external funcs for wasm module.
-    fn new(
-        ext: E,
+    /// 2) Creates wasm memory
+    /// 3) Runs `pre_execution_handler` to fill the memory before running instance.
+    /// 4) Instantiate external funcs for wasm module.
+    /// 5) Run instance setup starting at `entry_point` - wasm export function name.
+    fn execute<F, T>(
+        ext: &mut E,
         binary: &[u8],
         entries: BTreeSet<DispatchKind>,
         mem_size: WasmPageNumber,
```

Comment:
do we still need this arg?
we can take it from last allocation.

same for entries - why do we need them here?